tensorflow/core/kernels/data/experimental/sampling_dataset_op_test.cc

```diff
@@ -0,0 +1,733 @@
+
+namespace tensorflow {
+namespace data {
+namespace experimental {
+namespace {
+
+constexpr char kNodeName[] = "sampling_dataset";
+
+// Parameters for constructing a dataset that returns an ordered sequence
+// of numbers
+struct RangeDatasetParams {
+  int start;
+  int stop;
+  int step;
+};
+
+struct TakeDatasetParams {
+  int count;
+};
+
+class SamplingDatasetOpTest : public DatasetOpsTestBase {
+ protected:
+  // Creates a new `SamplingDataset` op kernel.
+  // Doesn't initialize the kernel's static parameters because they are inputs,
+  // not attributes.
+  Status CreateSamplingDatasetOpKernel(
+      const DataTypeVector& output_types,
+      const std::vector<PartialTensorShape>& output_shapes,
+      std::unique_ptr<OpKernel>* sampling_dataset_op_kernel) {
+    NodeDef node_def = test::function::NDef(
+        kNodeName, name_utils::OpName(SamplingDatasetOp::kDatasetType),
+        // Inputs
+        {SamplingDatasetOp::kInputDataset, SamplingDatasetOp::kRate,
+         SamplingDatasetOp::kSeed, SamplingDatasetOp::kSeed2},
+        // Attributes
+        {{SamplingDatasetOp::kOutputTypes, output_types},
+         {SamplingDatasetOp::kOutputShapes, output_shapes}});
+    TF_RETURN_IF_ERROR(CreateOpKernel(node_def, sampling_dataset_op_kernel));
+    return Status::OK();
+  }
+
+  // Creates an OpKernel context suitable for running a `SamplingDataset`
+  // kernel.
+  Status CreateSamplingDatasetContext(
+      OpKernel* const op_kernel,
+      gtl::InlinedVector<TensorValue, 4>* const inputs,
+      std::unique_ptr<OpKernelContext>* context) {
+    TF_RETURN_IF_ERROR(CheckOpKernelInput(*op_kernel, *inputs));
+    TF_RETURN_IF_ERROR(CreateOpKernelContext(op_kernel, inputs, context));
+    return Status::OK();
+  }
+
+  // Build a dataset that will return an ordered sequence of numbers in chunks
+  // of size `params.count`.
+  // Stuffs the returned dataset into a variant tensor.
+  Status MakeRangeAndTakeDatasetTensor(
+      const RangeDatasetParams& range_dataset_params,
+      const TakeDatasetParams& take_dataset_params,
+      Tensor* range_and_take_dataset_tensor) {
+    Tensor range_dataset_tensor;
+    Tensor start =
+        CreateTensor<int64>(TensorShape({}), {range_dataset_params.start});
+    Tensor stop =
+        CreateTensor<int64>(TensorShape({}), {range_dataset_params.stop});
+    Tensor step =
+        CreateTensor<int64>(TensorShape({}), {range_dataset_params.step});
+    TF_RETURN_IF_ERROR(MakeRangeDataset(start, stop, step, {DT_INT64},
+                                        {PartialTensorShape({})},
+                                        &range_dataset_tensor));
+
+    TF_RETURN_IF_ERROR(MakeTakeDataset(
+        range_dataset_tensor, take_dataset_params.count, {DT_INT64},
+        {PartialTensorShape({})}, range_and_take_dataset_tensor));
+    return Status::OK();
+  }
+};
+
+// Common parameters that every test case in this file shares
+struct TestCase {
```

Comment:
Please follow the style @feihugis and I converged on, having separate test case structs for different types of tests (https://github.com/tensorflow/tensorflow/pull/31105).

Reply:
@jsimsa Thanks for merging #31105! It will be helpful for us to refactor other tests!

Reply:
@frreiss More examples can be found [here](https://github.com/tensorflow/tensorflow/pull/31344).

Reply:
Thanks for the example code @feihugis. I have pushed a refactored version of the test case into the new format in commit edf1be3

Reply:
@frreiss Thanks for updating the test format! There may be some more changes for the format. Will let you know once it is finalized.

Reply:
@frreiss The changes in test formats are finished. An example can be found [here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/data/batch_dataset_op_test.cc). Please feel free to let me know if you have any question or need any help.

Reply:
Thanks @feihugis. I've restructured the tests in this PR to match the new format and checked the changes into commit d827bfa.